### PR TITLE
Feature/support constructor type to type

### PR DESCRIPTION
**Pull request by rufflian**
- Use `skipLibCheck` to ignore errors in libs unrelated to our code (fixes conflict within anchor with web3.js)
- Adds config for ts-jest
- Updates `typeToType` to support an input type of a class constructor and tests

### CODE REVIEW

File: src/instruction/__tests__/utils.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
+import { convertTypeToType } from "../utils";
+import {describe, expect, it} from "@jest/globals";
+
+describe("Type conversion with convertTypeToType", () => {
```
Comment (long-banana) [inline src/instruction/__tests__/utils.ts:4]: Can you please add a test for `(arg0: any) => any` to match the current functionality still works?

File: tsconfig.base.json [hunk 2]
```diff
@@ -15,6 +15,7 @@
     "resolveJsonModule": true,
     "composite": true,
     "baseUrl": ".",
+    "skipLibCheck": true,
```
Comment (long-banana) [inline tsconfig.base.json:18]: I would like to know more about the types of errors you were seeing

Comment (rufflian) [reply tsconfig.base.json:18]: Similar to this: https://stackoverflow.com/questions/65353067/error-ts6200-error-ts2403-definitions-of-the-following-identifiers-conflict-w but between anchor and web3.js

Comment (rufflian) [reply tsconfig.base.json:18]: I tried bumping versions but that didn't help 🤷‍♀️

File: src/instruction/__tests__/utils.ts (hunk 1 above)
Comment (rufflian) [reply src/instruction/__tests__/utils.ts:4]: @long-banana Do you mean something like this?
https://github.com/raindrop-studios/sol-kit/pull/11/commits/2c75d8a5d0c49365ff39a4a2a7e324d9bbcf97a7

File: tsconfig.base.json (hunk 2 above)
Comment (long-banana) [reply tsconfig.base.json:18]: I mean the actual errors. Are they just type definition errors? How many of them?

Comment (rufflian) [reply tsconfig.base.json:18]: On running `yarn build`
Screenshot_20221230_115045
Screenshot_20221230_115105

File: src/instruction/__tests__/utils.ts (hunk 1 above)
Comment (long-banana) [reply src/instruction/__tests__/utils.ts:4]: yup. perfect!

File: tsconfig.base.json (hunk 2 above)
Comment (long-banana) [reply tsconfig.base.json:18]: Ah yes. I'm familiar with that specific wall of errors. Thanks.

**Review by long-banana — APPROVED** [review]

**Pull request merged**